torch/jit/__init__.py

```diff
@@ -1513,7 +1513,6 @@ def __init__(self, optimize=None, _qualified_name=None, _compilation_unit=None,
                 self.__dict__['_c'] = torch._C.ScriptModule(_qualified_name, _compilation_unit, True)
 
             Module._Module__construct(self)
```

Comment:
The only point of this stupid `_module__construct` thing was to avoid setting "training". Let's just merge `__construct` and `__init__` in `nn.Module`. and call super's `__init__` here.

Reply:
What if someone makes a `script::Module` in C++? Should that also have a `training` attribute by default?